src/transformers/models/bert/modeling_tf_bert.py

```diff
@@ -740,6 +740,12 @@ def call(
         **kwargs,
     ) -> Union[TFBaseModelOutputWithPoolingAndCrossAttentions, Tuple[tf.Tensor]]:
 
+        # The received `past_key_values` is a tuple of 2 elements.
+        # The 1st element is `encoder_hidden_states`. The 2nd element is a tuple of `n_layers` elements,
+        # each element is a tuple of 4 tensors of shape (batch_size, n_heads, seq_len - 1, embed_size_per_head)
+        if type(past_key_values) == tuple and len(past_key_values) == 2 and type(past_key_values[1]) == tuple:
```

Comment:
This shoudn't be the case. `past_key_values` should **only** be the `2nd element` you describe above. @gante - did we miss this model in the `past/encoder_outputs` refactor?

Reply:
`TFBert` and `TFEncoderDecoder` were both updated ([PR](https://github.com/huggingface/transformers/pull/15944/files#diff-78a88286ed61e08098e9c0f2a9e75b0aea93d9b908d8e5da4b9b023bc27fc3e7)). The [test](https://github.com/huggingface/transformers/blob/b7fa1e3deebc3917965ad673e9fb61301416078c/tests/encoder_decoder/test_modeling_tf_encoder_decoder.py#L601) in question also doesn't have the `num_beams` nor the `sample` arguments in generate, meaning that it should be going through the updated `greedy_search`, which has the updated `past` format 🤔 

@ydshieh, can I ask you to confirm that `generate()` calls the new `_generate()` internally ([this one](https://github.com/huggingface/transformers/blob/master/src/transformers/generation_tf_utils.py#L565))? If not, it means that it is going through the old code, which may have some `past`-related issue (I made some ad hoc changes to comply with the new `past` format, but they may be incomplete.

Reply:
I can check the logic flow, will report back.

Reply:
- calls the new _generate() -> Yes
- then `return self.greedy_search` [here](https://github.com/huggingface/transformers/blob/a6271967c971cab5d882bba87c19406f30cdc7e4/src/transformers/generation_tf_utils.py#L1540)
- This line [model_inputs = self.prepare_inputs_for_generation](https://github.com/huggingface/transformers/blob/a6271967c971cab5d882bba87c19406f30cdc7e4/src/transformers/generation_tf_utils.py#L1920) gives `past_key_values`:
  - 1st step: None
  - 2nd step: a tuple of 2 elements (which shouldn't be)

It is probably this method [TFEncoderDecoderModel.prepare_inputs_for_generation](https://github.com/huggingface/transformers/blob/a6271967c971cab5d882bba87c19406f30cdc7e4/src/transformers/models/encoder_decoder/modeling_tf_encoder_decoder.py#L694) should be corrected, rather than in `TFBertModel`.

Considering it was me who worked on `TFEncoderDecoderModel`, I can fix it instead.
@gante, WDYT? If yes, do you have some hints/tips for me about the changes on the format you have done?

Reply:
So the problem should be related to [this line](https://github.com/huggingface/transformers/blame/a6271967c971cab5d882bba87c19406f30cdc7e4/src/transformers/models/encoder_decoder/modeling_tf_encoder_decoder.py#L706), and it probably implies I missed some updates 🙈 

I can look into it, it is almost surely related to a problem in my past PR

Reply:
Ok, thank you!